Assets/HoloToolkit/Input/Scripts/InputSources/InteractionInputSource.cs

```diff
@@ -909,12 +909,24 @@ private void InteractionManager_InteractionSourceUpdated(InteractionSourceUpdate
 
         private void InteractionManager_InteractionSourceReleased(InteractionSourceReleasedEventArgs args)
         {
-            InputManager.Instance.RaiseSourceUp(this, args.state.source.id, (InteractionSourcePressInfo)args.pressType);
+            var pressType = (InteractionSourcePressInfo)args.pressType;
+            if (args.state.source.kind == InteractionSourceKind.Hand)
+            {
+                pressType = InteractionSourcePressInfo.Select;
```

Comment:
Do we want to change the expected press type from the platform here, or should we handle this on the receiving end instead, checking for either `InteractionSourceKind.Hand` or `InteractionSourcePressInfo.Select`?

Reply:
See #1302

Reply:
That doesn't answer my question.

Reply:
Could we add checking for `InteractionSourceKind.Hand` here: https://github.com/Microsoft/MixedRealityToolkit-Unity/blob/master/Assets/HoloToolkit/Input/Scripts/Utilities/Managers/InputManager.cs#L425

instead of changing the press type in the event data?

Reply:
No I think where we're checking is the correct place.

We shouldn't have to know about what kind of source we're dealing with in the input manager.

Reply:
Not sure I agree, because I might want something different to happen in my app if it's hand vs controller vs voice, but this change is fine.

Does voice properly pass a Select type?

Reply:
It never did before.

There were comments and notes about this. But currently voice doesn't raise souce up/ down/ clicked so it doesn't matter.

Reply:
> But currently voice doesn't raise souce up/ down/ clicked so it doesn't matter.

I don't believe this is true.

Reply:
Confirmed: The "select" voice command both steals pointer focus (via OnInputDown) and enlarges the cube in the test scene (via OnInputClicked).

Reply:
`Select` is a protected voice command that's raising the click from the OS, correct?  I meant any custom voice commands in the MRTK via `SpeechInputSource`.

Reply:
Correct, talking about the context of `InteractionInputSource`.

Reply:
>. I wouldn't be surprised to see the "select" keyword sending in a type of None as well there

That wouldn't surprise me, smh, might just be the case.  What source is raised when we use the `Select` keyword? 'Cause it's not coming from `SpeechInputSource`.

Reply:
Right, it comes from `InteractionManager` and `GestureRecognizer`, into the Toolkit via `InteractionInputSource`.

Looks like it comes through with a source ID of 0 and a type of Voice. That led me to notice a bug though, because we don't pass the source kind through the events at the moment; it's only accessible through `TryGetSourceKind`, but Voice doesn't really have its own source, so that always returns the default of `Other`.

Reply:
I think this is a separate issue from the uGUI pointer stuff.  Any chance we could open a new issue and address it later with a new PR?

Reply:
The change here for `Hand` should likely also be made for `Voice`, so I don't believe that is a separate issue. If you test the uGUI stuff on HoloLens and use the "select" keyword, does it work as-is?

The Voice source type = `other` is definitely a separate issue.

Reply:
I'll be sure to check for both.

Reply:
> If you test the uGUI stuff on HoloLens and use the "select" keyword, does it work as-is?

I won't have a way to test this on device until Monday.  I took some vacation time. (Yes the irony about be still contributing even on vacation 😆 )

Reply:
I left a device reimaging, so I should be able to test tomorrow!

Ah! Enjoy your vacation :)

Reply:
I left a device reimaging, so I should be able to test tomorrow!

Ah! Enjoy your vacation :)

Reply:
I'll still be here 😆